libs/execution_context/ebpf_program.c

```diff
@@ -249,6 +249,9 @@ ebpf_program_create(ebpf_program_t** program)
         goto Done;
     }
 
+    ebpf_list_initialize(&local_program->links);
+    ebpf_lock_create(&local_program->links_lock);
```

Comment:
This is already happening in `ebpf_program_initialize()`

Reply:
ebpf_program_create then _ebpf_program_free and we call _ebpf_program_detach_links with a non-initialized value for program->links, triggering the verified failure.

Reply:
Should lines 309-310 be removed then so we don't need to initialize these twice?

Reply:
I think that flow is only present in the test code (`ebpf_program_create()` -> `_ebpf_program_free()`). In the EC code, everywhere, if call to `ebpf_program_create()` succeeds, then `ebpf_program_initialize()` is called.

Agree we should remove this code from ebpf_program_initialize() as it is moved here.